lib/builders.js

```diff
@@ -99,6 +130,128 @@ class JSONBuilder {
+        missingRepos.push(pkgName);
+      }
+    });
+
+    if (this.shouldNotifyLinks && !_.isEmpty(missingRepos)) {
+      console.log(
+        chalk.yellow(
+          `The following packages did not have a verifiable online LICENSE link: ${missingRepos
+            .toString()
+            .replace(',', '\n')}`
+        )
+      );
+    }
+    if (!_.isEmpty(empties)) {
+      console.log(
+        chalk.red(
+          `The following packages did not have a local LICENSE file OR a LICENSE file/text in cache: ${empties
+            .toString()
+            .replace(',', '\n')}`
+        )
+      );
+    }
+  }
+}
+
+class HTMLBuilder {
+  constructor() {
+    this.html = '';
+  }
+
+  buildHTML(licenseDefs) {
+    this.html = '<!DOCTYPE html><html><head></head><body>';
+    let linkOutput = '<div>';
+    let bodyOutput = '<div>';
+
+    _.forEach(licenseDefs, pkg => {
+      linkOutput += `<div>${this._makeLink(pkg)}</div>`;
+      bodyOutput += `<div style="margin-bottom: 20px">${this._makeLicenseEntry(pkg)}</div>`;
+    });
+
+    linkOutput += '</div>';
+    this.html += linkOutput;
+    bodyOutput += '</div>';
+    this.html += bodyOutput;
+    this.html += '</body>';
+    this.html += '</html>';
+  }
+
+  write() {
+    fs.writeFile('licenses.html', this.html, error => {
+      if (error) {
+        exit(error);
+      }
+
+      console.log(chalk.green('Wrote html licenses to licenses.html'));
+    });
+  }
+
+  _makeLink({ name, version }) {
+    return `<a href='#${name}'>${name} ${version}</a>`;
+  }
+
+  _makeLicenseEntry({ name, version, text }) {
+    return `<div><a id='${name}'>${name} ${version}</a></div><div>${text}</div>`;
+  }
+}
+
+class XLSXBuilder {
+  constructor() {
+    this.rows = [];
+    this.csvHeaders = ['name', 'version', 'licenses', 'text'];
+  }
+
+  add(pkg) {
+    this.rows.push({
+      name: pkg.name,
+      version: pkg.version,
+      licenses: pkg.licenses,
+      text: pkg.text
+    });
+  }
+
+  addCached(name, version, text, licenses) {
+    this.rows.push({ name, version, text, licenses });
+  }
+
+  addEmpty(name, version) {
+    this.rows.push({ name, version, text: '', licenses: '' });
+  }
+
+  setHeaders(headerArray) {
+    this.csvHeaders = headerArray;
+  }
+
+  convertToRows(licenseDefs) {
+    this.rows = _.map(licenseDefs, licenseDef => {
+      return {
+        name: licenseDef.name || '',
+        version: licenseDef.version || '',
+        licenses: licenseDef.licenses || '',
+        text: licenseDef.text || ''
+      };
+    });
+  }
+
+  write(name) {
+    const ws = xlsx.utils.json_to_sheet(this.rows, {
+      header: this.csvHeaders
+    });
+    const wb = xlsx.utils.book_new();
+    xlsx.utils.book_append_sheet(wb, ws, 'sheet1');
+    const outputName = name ? name : 'output';
+    xlsx.writeFile(wb, `${outputName}.csv`, { bookType: 'csv' });
+  }
 }
 
-module.exports = { JSONBuilder, TextBuilder };
+module.exports = { JSONBuilder, TextBuilder, XLSXBuilder };
```

Comment:
Need to export the HTML builder?

Reply:
I may end up moving it out or something. Either way it will come in the next PR, which will be the  HTML generator